File: javascript/packages/core/src/extract-ruby-options.ts
import type { LexOptions } from "./parser-options.js"

export interface ExtractRubyOptions extends LexOptions {
  semicolons?: boolean
  comments?: boolean
  preserve_positions?: boolean
  custom_tags?: boolean
}

export const DEFAULT_EXTRACT_RUBY_OPTIONS: ExtractRubyOptions = {
  semicolons: true,
  comments: false,
  preserve_positions: true,
  custom_tags: false,
}

File: java/org/herb/ExtractRubyOptions.java
package org.herb;

public class ExtractRubyOptions {
  private boolean semicolons = true;
  private boolean comments = false;
  private boolean preservePositions = true;

  public ExtractRubyOptions() {}

  public ExtractRubyOptions semicolons(boolean value) {
    this.semicolons = value;
    return this;
  }

  public boolean isSemicolons() {
    return semicolons;
  }

  public ExtractRubyOptions comments(boolean value) {
    this.comments = value;
    return this;
  }

  public boolean isComments() {
    return comments;
  }

  public ExtractRubyOptions preservePositions(boolean value) {
    this.preservePositions = value;
    return this;
  }

  public boolean isPreservePositions() {
    return preservePositions;
  }

  public static ExtractRubyOptions create() {
    return new ExtractRubyOptions();
  }
}
